### ISSUE TITLE
setters for ApiKey

### ISSUE THREAD

**Issue by idlenexusgaming** [tarioch/eveapi-fetcher-bundle#21]
Would you be willing to add setters to your ApiKey.php entity class for keyID and vCode? I've had a lot of trouble trying to extend your class to make it work in my bundle.

**Comment by tarioch** [comment]
Could you explain why you would need setters instead of the constructor? I generally prefer to keep the "mandatory" stuff as constructor arguments.

**Comment by idlenexusgaming** [comment]
based on my limited exposure, I don't really know what I need. Can you please suggest how to insert into the apikey table using your ApiKey class?

**Comment by tarioch** [comment]
If $em is the entity manager for eveapi:

$em->persist(new ApiKey($keyId, $vCode));

**Comment by tarioch** [comment]
and for wiring the em you can use something like "doctrine.orm.eveapi_entity_manager"

**Comment by idlenexusgaming** [comment]
Hey thanks again. I did try that but I realized where I was making the mistake. When I set $em = $this->get('doctrine')->getManager('eveapi')->getRepository(...) and then later ran $em->flush(); it would through errors because flush doesn't live in getRepository. I split it up to two lines and all is well. Everything is working perfectly. 

One last thing to point out. The eve api supports updating the vcode but not the key id. There is no way to setVcode to update it and trying to use the construct to add the key to the db throws the primary key restriction error. It either needs an id column to replace the primary key or there needs to be a way to update the vcode. So maybe a setter for vcode is necessary after all?
